app/services/task_list_creator.rb

```diff
@@ -11,12 +11,23 @@ def call(project)
   end
 
   private def create_task_list_from_workflow(project, workflow)
-    workflow["sections"].each_with_index do |workflow_section, index|
-      section = Section.create(title: workflow_section["title"], order: index, project: project)
+    workflow.fetch("sections", []).each_with_index do |workflow_section, index|
```

Comment:
I can see where you're coming from in that we're confident (or should be confident) `sections` will always exist because it's enforced by the validator. To me that would mean we don't need to have a default here, because any time we receive a workflow without a section something has gone wrong somewhere else (for whatever reason) and it makes sense to catch it.

I don't think it needs a specific error type though. `WorkflowFileInvalid` is what the validator does, this is a straight case of `KeyError` in that our code requires one to be able to continue and it doesn't exist.

Reply:
👍 I've dropped the defaults so if we do get into that state there will be a `KeyError`